AuthoringToolTest/PresentationLogic/ModalDialog/ModalDialogInputFieldsFactoryUt.cs

```diff
@@ -9,6 +15,277 @@ namespace AuthoringToolTest.PresentationLogic.ModalDialog;
 
 public class ModalDialogInputFieldsFactoryUt
 {
+    [Test]
+    public void GetCreateLearningElementInputFields_Test()
+    {
+        var systemUnderTest = new ModalDialogInputFieldsFactory();
+
+        for (int contentFileTypeTest = 0; contentFileTypeTest < 8; contentFileTypeTest++)
+        {
+            string name = "foo";
+            string contentFileExtension =
+                helper_GetCreateLearningElementInputFields_ReturnsExpectedFileExtension(contentFileTypeTest);
+
+            var contentType =
+                helper_GetCreateLearningElementInputFields_ReturnsExpectedContentType(contentFileTypeTest);
+
+
+
+            string spaceName = "skazzle";
+
+            LearningContentViewModel? dragAndDropLearningContent =
+                new LearningContentViewModel(
+                    name,
+                    contentFileExtension,
+                    new byte[] { }
+                );
+            
+            try
+            {
```

Comment:
Niemals try/catch in Testmethoden, weil die Asserts Exceptions werfen. Stattdessen Testmethode in positiven und negativen Test aufteilen, und im negativen Test in dem du die Exception erwartest mit `Assert.That(()=>foo.bar(), Throws.TypeOf<WhateverException>())` abfangen. Vorteil: Der Assert schlägt fehl, wenn die Exception nicht fliegt oder die falsche Exception fliegt. https://docs.nunit.org/articles/nunit/writing-tests/constraints/ThrowsConstraint.html

Reply:
das problem dabei ist doch dann eigentlich, dass in der zu testenden datei eine standard exception geworfen wird, was zu dem problem führt, dass selbst wenn eine andere exception geworfen wird hier doch der test erfolgreich ist, da andere exceptions von der standard exception abgeleitet werden wenn ich mich nicht irre?

Reply:
Das Problem bei deinem Lösungsvorschlag wäre dann doch ebenfalls dass der Assert erfolgreich ist, falls irgendeine andere exception geworfen wird, da in der zu testenden Funktion des Autorentools eine Standard Exception geworfen wird welche doch als parent zu sämtlichen anderen Exceptions dient.

`Assert.Throws<Exception>(()=>systemUnderTest.GetCreateLearningElementInputFields(dragAndDropLearningContent,
                spaceName), "Can not map the file extension 'abc' to a ContentType ");`
                
                das wäre mein Lösungsvorschlag dieser stellt sicher, dass die Standard Exception geworfen wird und überprüft ebenfalls die Exception message welche geworfen wird